### PR TITLE
load: don't block when worker cancel

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
Don't block on send jobs to jobQueue when context is canceled.

### What is changed and how it works?
Add select statement on jobQueue

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### Releasae note
- Fix the issue that the load unit might block when context canceled.

### CODE REVIEW

File: loader/loader.go [hunk 1]
```diff
@@ -364,7 +364,10 @@ func (w *Worker) dispatchSQL(ctx context.Context, file string, offset int64, tab
 			}
 			lastOffset = cur
 
-			w.jobQueue <- j
+			select {
```
Comment (overvenus) [inline loader/loader.go:367]: What about load/loader.go:250? Should we `select ctx.Done` as well?

Comment (3pointer) [reply loader/loader.go:367]: yes, we should add it too, I'll fix it.

**Comment by GMHDBJD** [comment]
Has this issue been fixed in #1276 after 2.0.1?

**Comment by lance6716** [comment]
> Has this issue been fixed in #1276 in 2.0.1?

that pr makes the receiving end of channel could exit, but the sending end may block because the channel is full.

**Comment by lance6716** [comment]
replaced by https://github.com/pingcap/dm/pull/1276

**Pull request closed without merging**